lib/galaxy/datatypes/isa.py

```diff
@@ -0,0 +1,520 @@
+"""
+ISA datatype
+
+See https://github.com/ISA-tools
+
+"""
+
+from __future__ import print_function
+
+import json
+import logging
+import os
+import os.path
+import re
+import shutil
+import sys
+import tarfile
+import tempfile
+import zipfile
+from cgi import escape
+from io import BytesIO
+from json import dumps  # noqa: F401
+
+# Imports isatab after turning off warnings inside logger settings to avoid pandas warning making uploads fail.
+logging.getLogger("isatools.isatab").setLevel(logging.ERROR)
+from isatools import isajson
+from isatools import isatab
+
+from galaxy import model
+from galaxy import util
+from galaxy.datatypes import data
+from galaxy.util.sanitize_html import sanitize_html
+
+# CONSTANTS {{{1
+################################################################
+
+# Main files regex
+JSON_FILE_REGEX = re.compile(r"^.*\.json$", flags=re.IGNORECASE)
+INVESTIGATION_FILE_REGEX = re.compile(r"^i_\w+\.txt$", flags=re.IGNORECASE)
+
+# The name of the ISA archive (compressed file) as saved inside Galaxy
+ISA_ARCHIVE_NAME = "archive"
+
+# Archives types
+_FILE_TYPE_PREFIX = {
+    "\x1f\x8b\x08": "gz",
+    "\x42\x5a\x68": "bz2",
+    "\x50\x4b\x03\x04": "zip"
+}
+_MAX_LEN_FILE_TYPE_PREFIX = max(len(x) for x in _FILE_TYPE_PREFIX)
+_FILE_TYPE_REGEX = re.compile("(%s)" % "|".join(map(re.escape, _FILE_TYPE_PREFIX.keys())))
+
+# Set max number of lines of the history peek
+_MAX_LINES_HISTORY_PEEK = 11
+
+# Configure logger {{{1
+################################################################
+
+logger = logging.getLogger(__name__)
+ch = logging.StreamHandler(sys.stdout)
+formatter = logging.Formatter("%(name)s %(levelname)s %(asctime)s %(message)s")
+ch.setFormatter(formatter)
+logger.handlers = []
+logger.propagate = False
+logger.addHandler(ch)
+logger.setLevel(logging.ERROR)
```

Comment:
I sort of understand the logging above - does this logging need to be configured also?

Reply:
Sorry @jmchilton , I fail to understand your remark. Could you elaborate?